oss-internship-2020/libtiff/example/main_sandboxed.cc

```diff
@@ -0,0 +1,305 @@
+  }
+
+  return absl::InternalError(absl::StrCat(
+      "Pixel ", pixel, " did not match expected results.\n",
+      "Got R=", TIFFGetR(rgba), " (expected ", min_red, "..", max_red,
+      "), G=", TIFFGetG(rgba), " (expected ", min_green, "..", max_green,
+      "), B=", TIFFGetB(rgba), " (expected ", min_blue, "..", max_blue, "), A=",
+      TIFFGetA(rgba), " (expected ", min_alpha, "..", max_alpha, ")\n"));
+}
+
+}  // namespace
+
+std::string GetFilePath(const std::string& dir, const std::string& filename) {
+  return sandbox2::file::JoinPath(dir, "test", "images", filename);
+}
+
+std::string GetFilePath(const std::string filename) {
+  std::string cwd = sandbox2::file_util::fileops::GetCWD();
+  auto find = cwd.rfind("build");
+
+  std::string project_path;
+  if (find == std::string::npos) {
+    LOG(ERROR) << "Something went wrong: CWD don't contain build dir. "
+               << "Please run tests from build dir or send project dir as a "
+               << "parameter: ./sandboxed /absolute/path/to/project/dir \n";
+    project_path = cwd;
+  } else {
+    project_path = cwd.substr(0, find);
+  }
+
+  return sandbox2::file::JoinPath(project_path, "test", "images", filename);
+}
+
+absl::Status LibTIFFMain(const std::string& srcfile) {
+  // without addDir to sandbox. to add dir use
+  // sandbox(absolute_path_to_dir, srcfile) or
+  // sandbox(absolute_path_to_dir). file and dir should be exists.
+  // srcfile must also be absolute_path_to_file
+  TiffSapiSandbox sandbox("", srcfile);
+
+  // initialize sapi vars after constructing TiffSapiSandbox
+  sapi::v::UShort h, v;
+  sapi::StatusOr<TIFF*> status_or_tif;
+  sapi::StatusOr<int> status_or_int;
+  sapi::StatusOr<tmsize_t> status_or_long;
+  absl::Status status;
+
+  status = sandbox.Init();
+
+  SAPI_RETURN_IF_ERROR(sandbox.Init());
+
+  TiffApi api(&sandbox);
+  sapi::v::ConstCStr srcfile_var(srcfile.c_str());
+  sapi::v::ConstCStr r_var("r");
+
+  SAPI_ASSIGN_OR_RETURN(
+      status_or_tif, api.TIFFOpen(srcfile_var.PtrBefore(), r_var.PtrBefore()));
+
+  sapi::v::RemotePtr tif(status_or_tif.value());
+  if (!tif.GetValue()) {
+    // tif is NULL
+    return absl::InternalError(absl::StrCat("Could not open ", srcfile));
+  }
+
+  SAPI_ASSIGN_OR_RETURN(status_or_int,
+                        api.TIFFGetField2(&tif, TIFFTAG_YCBCRSUBSAMPLING,
+                                          h.PtrBoth(), v.PtrBoth()));
+  if (status_or_int.value() == 0 || h.GetValue() != 2 || v.GetValue() != 2) {
+    return absl::InternalError("Could not retrieve subsampling tag");
+  }
+
+  SAPI_ASSIGN_OR_RETURN(status_or_long, api.TIFFTileSize(&tif));
+  if (status_or_long.value() != 24576) {
+    return absl::InternalError(
+        absl::StrCat("tiles are ", status_or_long.value(), " bytes\n"));
+  }
+  tsize_t sz = status_or_long.value();
+
+  sapi::v::Array<uint8_t> buffer_(sz);
+  SAPI_ASSIGN_OR_RETURN(
+      status_or_long, api.TIFFReadEncodedTile(&tif, 9, buffer_.PtrBoth(), sz));
```

Comment:
The `9` sounds random - not sure if this is from the existing test case, but maybe add a quick comment what this line is supposed to do and why there is this 9.